nailed-it/New Group/PolishLibraryViewController.swift

```diff
@@ -193,55 +242,111 @@ extension PolishLibraryViewController: CZPickerViewDelegate, CZPickerViewDataSou
     }
 
     func numberOfRows(in pickerView: CZPickerView!) -> Int {
-        return self.brands.count
+        if self.sortFilter == "filter" {
+            return self.brands.count
+        } else {
+            return self.sortingOptions.count
+        }
     }
 
     func numberOfRowsInPickerView(pickerView: CZPickerView!) -> Int {
-        return self.brands.count
+        if self.sortFilter == "filter" {
+            return self.brands.count
+        } else {
+            return self.sortingOptions.count
+        }
     }
 
     func czpickerView(_ pickerView: CZPickerView!, titleForRow row: Int) -> String! {
-        return self.brands[row]
-    }
+        if self.sortFilter == "filter" {
+            return self.brands[row]
+        } else {
+            return self.sortingOptions[row]
+        }
 
-    func czpickerView(_ pickerView: CZPickerView!, didConfirmWithItemAtRow row: Int){
-        self.navigationController?.setNavigationBarHidden(true, animated: true)
     }
 
     func czpickerViewDidClickCancelButton(_ pickerView: CZPickerView!) {
         self.navigationController?.setNavigationBarHidden(false, animated: true)
     }
+    
+    func czpickerView(_ pickerView: CZPickerView!, didConfirmWithItemAtRow row: Int){
```

Comment:
Can you give me a quick rundown of what this is doing?

Reply:
Yep, this is a pop up view that allows users to sort/filter. It's from the CZPicker cocoapod. 

<img width="245" alt="screen shot 2017-10-22 at 3 03 01 pm" src="https://user-images.githubusercontent.com/6549873/31866819-1a55c964-b73a-11e7-9e15-6c6c19779422.png">
<img width="240" alt="screen shot 2017-10-22 at 3 02 55 pm" src="https://user-images.githubusercontent.com/6549873/31866820-1a6ac8be-b73a-11e7-9b39-345402196483.png">